config/__init__.py

```diff
@@ -307,7 +307,10 @@ def _suppress_call(
 
         return None
 
-    def get(self, key: str, default: Union[list, str, bool, int] = None) -> Union[list, str, bool, int]:
+    def get(
+        self, key: str,
+        default: Union[list, str, bool, int, None] = None
+    ) -> Union[list, str, bool, int, None]:
```

Comment:
```suggestion
        default: list | str | bool | int | None = None
    ) -> list | str | bool | int | None:
```

Or something that defines a TypeVar that can be all of those, then accept it as the default and use it as the returntype, which would be something like

```py
T = typing.TypeVar('T', list, str, bool, int, None)
def get(self, key: str, default: T) -> T:
    ...
```
the returntype here would then become defined by default, so it may break things elsewhere

Reply:
I'll go with the Python 3.10-esque `|` version, because:

```python
getT = TypeVar('getT', list, str, bool, int, None)

    def get(
        self, key: str,
        default: getT = None
    ) -> getT:
```
yields:
```python
0$ mypy --follow-imports normal config/__init__.py
config\__init__.py:315: error: Incompatible default for argument "default" (default has type "None", argument has type "List[Any]")  [assignment]
config\__init__.py:315: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
config\__init__.py:315: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
config\__init__.py:315: error: Incompatible default for argument "default" (default has type "None", argument has type "str")  [assignment]
config\__init__.py:315: error: Incompatible default for argument "default" (default has type "None", argument has type "bool")  [assignment]
config\__init__.py:315: error: Incompatible default for argument "default" (default has type "None", argument has type "int")  [assignment]
config\__init__.py:332: error: Cannot determine type of "a_str"  [has-type]
config\__init__.py:335: error: Cannot determine type of "a_bool"  [has-type]
config\__init__.py:338: error: Cannot determine type of "an_int"  [has-type]
Found 7 errors in 1 file (checked 1 source file)
```
It's the default value on `default` that causes that first line.  And in this case we do want to set that.